src/helpers/xcode.ts

```diff
@@ -0,0 +1,61 @@
+import fs from 'fs'
+
+import { info, UploadLogger } from '../helpers/logger'
+import {
+  XcodeCoverageFileReport,
+  XcodeCoverageReport,
+} from '../types'
+import { isProgramInstalled, runExternalProgram } from "./util"
+
+export async function generateXcodeCoverageFiles(archivePath: string): Promise<string> {
+  if (!isProgramInstalled('xcrun')) {
+      throw new Error('xcrun is not installed, cannot process files')
+  }
+  info('Running xcode coversion...')
+
+  const coverage: XcodeCoverageReport = {}
+  const report = { coverage: coverage }
+  const path = archivePath[0]
+
+  getFileList(path).forEach(repoFilePath => {
+    UploadLogger.verbose(`Converting ${repoFilePath}...`)
+    const coverageInfo = getCoverageInfo(path, repoFilePath)
+    const coverageJson = convertCoverage(coverageInfo)
+    report.coverage[repoFilePath] = coverageJson
+  })
+
+  let pathFilename = path.split('/').pop()
+  if (pathFilename) {
+    pathFilename = pathFilename.split('.xcresult')[0]
+  }
+  const filename = `./coverage-report-${pathFilename}.json`
+  UploadLogger.verbose(`Writing coverage to ${filename}`)
+  fs.writeFileSync(filename, JSON.stringify(report))
```

Comment:
```suggestion
  await fs.writeFile(filename, JSON.stringify(report))
```

Be sure to modify the import statement to load the `fs/promises` module instead of the `fs`.

Reply:
Out of curiosity, why is this necessary?

Reply:
There are very strong benefits and very few drawbacks to using an asynchronous version of a function over its synchronous counterpart.

`writeFile` is asynchronous and, as a result, does not block the execution of the program. Node.js can schedule and run other parts of the program _while_ the file is being written to. For the uploader in particular, this may mean very little in its current state; however, if other areas of it were written to take serious advantage of asynchronous functionality, further asynchrony would be synergetic with the overall performance of the application. Function calls could weave together and through each other.